src/config/settings.js

```diff
@@ -38,17 +38,37 @@ function load() {
   let settings;
   try {
     settings = JSON.parse(rawdata);
-  } catch(_) {
-    settings = {}
+  } catch (_) {
+    save(defaultSettings);
+    return defaultSettings;
   }
-  // clone settings because we check to see if the returned value is different
   let cfg = updateSettings(Object.assign({}, settings));
   if (!isDeepEqual(cfg, settings)) {
     save(cfg);
   }
   return cfg;
 }
 
+/**
+ * Verify that the loaded version of kubernetes, if specified, is in the current list of supported versions.  Throw an exception if not.
+ * @param{Object} settings
+ */
+
+function verifyLocalSettings(settings) {
+  const rawData = fs.readFileSync(process.cwd() + '/src/generated/versions.json');
+  const supportedVersions = JSON.parse(rawData.toString());
+  const proposedVersion = settings.kubernetes?.version;
+  if (proposedVersion && supportedVersions.indexOf(proposedVersion) < 0) {
+    const header = "Error in saved settings.json file";
+    const message = `Proposed kubernetes version ${proposedVersion} not supported`;
+    const {dialog} = require('electron');
+    dialog.showErrorBox(header, message);
+    throw new InvalidStoredSettings(message);
+    //TODO: Drop next line
+    // throw new Error(`Proposed kubernetes version ${proposedVersion} not supported`);
```

Comment:
This is unreachable code anyways; remove?

Reply:
long gone